src/test/java/seedu/siasa/logic/commands/contact/EditContactCommandTest.java

```diff
@@ -95,6 +123,19 @@ public void execute_filteredList_success() {
         String expectedMessage = String.format(EditContactCommand.MESSAGE_EDIT_CONTACT_SUCCESS, editedContact);
 
         Model expectedModel = new ModelManager(new Siasa(model.getSiasa()), new UserPrefs());
+
+        // Update contact policies
+        List<Policy> policyList = expectedModel.getFilteredPolicyList();
+        ArrayList<Pair<Policy, Policy>> policiesToBeUpdated = new ArrayList<>();
+        for (Policy p : policyList) {
+            if (p.getOwner().equals(expectedModel.getFilteredContactList().get(0))) {
+                policiesToBeUpdated.add(new Pair<>(p, newPolicyWtihNewOwner(p, editedContact)));
+            }
+        }
+        for (Pair<Policy, Policy> pair : policiesToBeUpdated) {
+            expectedModel.setPolicy(pair.getKey(), pair.getValue());
+        }
```

Comment:
Could we have a private method to update the expectedModel for DRY purposes?

Reply:
Private method extracted as of commit b9207c69fb0f2cf850d6036f8f1bfae9dd474fb8